src/components/AppLayout/Submenu /index.tsx

```diff
@@ -0,0 +1,84 @@
+import { ReactNode, useState, ComponentProps, PropsWithChildren } from "react";
+import clsx from "clsx";
+import { LegacyChevronRight1pxIcon } from "@deriv/quill-icons";
+import { Text } from "../../Text";
+import "./Submenu.scss";
+
+type TSubmenu = {
+    icon: ReactNode;
+    label: string;
+    labelSize?: ComponentProps<typeof Text>["size"];
+    submenuContent: ReactNode;
+    submenuClassName?: ComponentProps<"div">["className"];
+    className?: ComponentProps<"button">["className"];
+};
+
+/**
+ * Represents a submenu component with expandable/collapsible functionality.
+ * This component displays a button that, when clicked, toggles the visibility
+ * of a submenu panel. The submenu can contain any ReactNode elements provided
+ * through props and has customizable text and icon components.
+ *
+ * @component
+ * @param {ReactNode} props.icon - The icon displayed in the button that toggles the submenu.
+ * @param {string} props.label - The label text displayed next to the icon in the toggle button.
+ * @param {string} [props.labelSize="md"] - The size of the label text, defaults to "md".
+ * @param {ReactNode} props.submenuContent - The content displayed inside the submenu when it is open.
+ * @param {string} [props.submenuClassName] - Optional custom class name for styling the submenu container.
+ * @param {string} [props.className] - Optional custom class name for styling the toggle button.
+ * @param {ReactNode} props.children - The children nodes provided to the submenu panel, which are displayed below the submenuContent.
+ * @returns {JSX.Element} The rendered Submenu component with toggle functionality.
+ */
+export const Submenu = ({
```

Comment:
```suggestion
export const SubMenu = ({
```
also for the rest please 🙏